### PR TITLE
fix rwinrm command

### PR DESCRIPTION
**Pull request by MikaelSmith**
The `rwinrm` command used an API that appears to be absent. Update it to
use the current `WinRM::Connection` API.

### CODE REVIEW

**Comment by MikaelSmith** [comment]
This changes the prompt, as I wasn't quite sure how to recreate getting the full cmd prompt with the new API. Overall the implementation's cleaner, but it'd be nice to give some of the context of the previous prompt.

I'm also tempted to implement a powershell version of this.

File: bin/rwinrm [hunk 1]
```diff
@@ -19,7 +19,6 @@
 
 $LOAD_PATH.push File.expand_path('../../lib', __FILE__)
 
-require 'readline'
```
Comment (MikaelSmith) [inline bin/rwinrm]: This was redundant and caused warnings to be printed.

**Comment by sneal** [comment]
Nice, I had forgotten about that CLI. I think ultimately we want to take advantage of the new PowerShell goodness in V2 of this gem. I'll merge this since working software is always > than broken software. Thanks!

**Comment by MikaelSmith** [comment]
The command-line handling in this PR is pretty bad, but at least it works.

**Comment by joshcooper** [comment]
Thanks @MikaelSmith!

@sneal @mwrock The `rwinrm` CLI uses a `cmd` shell to send stdin. Is there are equivalent method when using `powershell`? I tried sending a `WriteStdin` message, but I get back:

```
#<WinRM::WinRMWSManFault: [WSMAN ERROR CODE: 2150858817]: <f:WSManFault Code='2150858817' Machine='xxx' xmlns:f='http://schemas.microsoft.com/wbem/wsman/1/wsmanfault'><f:Message>The Windows Remote Shell cannot process the request. The SOAP packet contains an element Stream that is invalid. Retry the request with the correct XML element. </f:Message></f:WSManFault>>
```

But it looks ok to me:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<env:Envelope
    xmlns:xsd="http://www.w3.org/2001/XMLSchema"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xmlns:env="http://www.w3.org/2003/05/soap-envelope"
    xmlns:a="http://schemas.xmlsoap.org/ws/2004/08/addressing"
    xmlns:b="http://schemas.dmtf.org/wbem/wsman/1/cimbinding.xsd"
    xmlns:n="http://schemas.xmlsoap.org/ws/2004/09/enumeration"
    xmlns:x="http://schemas.xmlsoap.org/ws/2004/09/transfer"
    xmlns:w="http://schemas.dmtf.org/wbem/wsman/1/wsman.xsd"
    xmlns:p="http://schemas.microsoft.com/wbem/wsman/1/wsman.xsd"
    xmlns:rsp="http://schemas.microsoft.com/wbem/wsman/1/windows/shell"
    xmlns:cfg="http://schemas.microsoft.com/wbem/wsman/1/config">
  <env:Header>
    <a:To>http://xxx:5985/wsman
    </a:To>
    <a:ReplyTo>
      <a:Address mustUnderstand="true">http://schemas.xmlsoap.org/ws/2004/08/addressing/role/anonymous
      </a:Address>
    </a:ReplyTo>
    <w:MaxEnvelopeSize mustUnderstand="true">153600
    </w:MaxEnvelopeSize>
    <a:MessageID>uuid:80735E39-26AB-47A2-A1EA-4A4A9017D6B6
    </a:MessageID>
    <p:SessionId mustUnderstand="false">uuid:90E59B25-8775-4213-9D86-1D7F4D78887F
    </p:SessionId>
    <w:Locale xml:lang="en-US" mustUnderstand="false"/>
    <p:DataLocale xml:lang="en-US" mustUnderstand="false"/>
    <w:OperationTimeout>PT60S
    </w:OperationTimeout>
    <w:ResourceURI mustUnderstand="true">http://schemas.microsoft.com/powershell/Microsoft.PowerShell
    </w:ResourceURI>
    <a:Action mustUnderstand="true">http://schemas.microsoft.com/wbem/wsman/1/windows/shell/Send
    </a:Action>
    <w:SelectorSet>
      <w:Selector Name="ShellId">13F742A5-C35F-48E8-B0F9-C398C1224345
      </w:Selector>
    </w:SelectorSet>
  </env:Header>
  <env:Body>
    <rsp:Send>
      <rsp:Stream Name="stdin" CommandId="13921B43-6FE6-43D6-B455-E6EF1678A1A1">eyJtZXNzYWdlIjoiaGkifQ==
      </rsp:Stream>
    </rsp:Send>
  </env:Body>
</env:Envelope>
```
Is this because the `CreatePipeline` message is hardcoded to send `NoInput=true`?

**Comment by mwrock** [comment]
Thats at least part of it. The powershell shell implementation is very much "optimized" to sending a command (or pipeline) and reading the response. We didn't attempt to handle streaming stdin. Because the powershell uses the PSRP protocol, it would need to manage the PSRP message types that support stdin. There may be a better way to articulate that, but the best starting point is to dive into the PSRP spec to research what that would look like.

**Pull request closed without merging**